### PR TITLE
Set up GH Actions

### PR DESCRIPTION
**Pull request by DivineDominion**
Experiment with GH Actions instead of Travis

### CODE REVIEW

**Comment by DivineDominion** [comment]
Oh boy this is frustrating and weird.

- Caching resources
  - https://sarunw.com/posts/caching-dependencies-in-github-actions/
  - https://github.com/actions/cache
  - https://github.com/ReactiveCocoa/ReactiveCocoa/blob/master/.github/workflows/master.yml
- Build actions from the marketplace were rather limited
  - https://github.com/marketplace/actions/xcode-build
  - https://github.com/marketplace/actions/xcode-test
  - https://github.com/devbotsxyz/carthage-bootstrap/

**Comment by DivineDominion** [comment]
So!

This is intended to replace Travis, which I couldn't get to run anymore. this PR is blocked by Travis, too,  you see.

What do y'all think about dropping our Travis build in favor of GitHub Actions? I have barely figured out how to make this work, and we're currently not testing Xcode 11 or Swift 4.2 builds with this. :/ 

Help would be appreciated (and much needed)

**Comment by mjarvis** [comment]
@DivineDominion Thanks for all this work. I've put aside some time for myself tomorrow to look at this in detail.

**Comment by DivineDominion** [comment]
@mjarvis Did you have a change to look at this? What do you think of it?

**Comment by DivineDominion** [comment]
@dani-mp @mjarvis Pinging for feedback if you think GH Actions are the way to go and f you spot any errors in the setup. I am considering to drop the annoying-to-figure-out cache setup and live with builds taking 10mins in total instead when we use this in the main ReSwift repo.

File: .github/workflows/swift.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+name: Swift
```
Comment (dani-mp) [inline .github/workflows/swift.yml]: I would name this workflow with a more descriptive name. What does it really do?

**Review by dani-mp — COMMENTED** [review]
Left a minor comment, otherwise 💯

File: .github/workflows/swift.yml (hunk 1 above)
Comment (DivineDominion) [reply .github/workflows/swift.yml]: Didn't even notice this tbh :)  Renamed the workflow but am open to suggestions.

Comment (DivineDominion) [reply .github/workflows/swift.yml]: image

Oh well this does look more informative, though a bit verbose perhaps

**Review by dani-mp — APPROVED** [review]
Awesome!

**Pull request merged**